### ISSUE TITLE
Three steps in action can't work

### ISSUE THREAD

**Issue by nothingshiru** [photopea/photopea#3005]
These 3 steps in action don't work in photopea.

1. Delete mask
2. hide smart filter (hide the first smart filter) 
3. show smart filter (show the first smart filter) 
![OTS4%3{RXH4$_QR 4 7VB`I](https://user-images.githubusercontent.com/79258245/110888132-85b29000-8326-11eb-9bda-f02003ea66be.png)

**Comment by photopea** [comment]
Hi, could you share the .ATN file with us here?

**Comment by nothingshiru** [comment]
[test.zip](https://github.com/photopea/photopea/files/6129175/test.zip)

**Comment by nothingshiru** [comment]
These actions is not working or working properly either. 

1. Delete the first smart filter. 
![Delete first smart filter](https://user-images.githubusercontent.com/79258245/111057682-59347a80-84c4-11eb-8584-989080df5d0a.jpg)

2. Clear smart filters. 
![Clear smart filter](https://user-images.githubusercontent.com/79258245/111057692-710bfe80-84c4-11eb-8acd-92dd01b0406e.jpg)

3. Hide all smart filters. 
![Hide all smart filter](https://user-images.githubusercontent.com/79258245/111057726-ac0e3200-84c4-11eb-9dea-d947af5a6d90.jpg)

4. Drag smart filter to another layer. 
![Drag smart filter](https://user-images.githubusercontent.com/79258245/111057737-bdefd500-84c4-11eb-9cd0-f5e5fd9d8e65.jpg)

5. Copy smart filter to another layer. 
![Copy smart filter](https://user-images.githubusercontent.com/79258245/111057791-1d4de500-84c5-11eb-8a55-0b82145b8547.jpg)

6. Delete a layer when not on it. 
![Delete layer when not on it](https://user-images.githubusercontent.com/79258245/111057780-fb546280-84c4-11eb-974b-80386f45f848.jpg)

**Comment by photopea** [comment]
There are over 1000 kinds of actions, that Photopea can not play correctly. We are aware of them. We would rather focus  on those that people actually need for a specific task, instead of those people randomly find not working.

Do you really need these?

**Comment by nothingshiru** [comment]
Yes I need these except "Again".

**Comment by photopea** [comment]
I fixed some actions, but others are hard to fix.

The mechanism of actions is not very clear to me. The action e.g. mentions a layer A, or a current layer. But what if there are multiple layers with the name "A", or no layer at all. When you delete a current layer with name "A" in Photoshop, does it create an action of deleting a layer with the name "A", or a selected layer? Can you specify what way should the action be recorded in Photoshop?

**Comment by nothingshiru** [comment]
If there were multiple layers with a same name, the steps in action will be executed on the first one of them. The first one is lowest one. 

If there are no any layers with the name, a pop-up window appears and says "This step is unusable, would you like to skip this step or stop this action?". 

If I delete current layer, the step recorded is delete selected layer. There is nothing to do with the name of layer. 

I discovered that the action dragging a smart to another layer doesn't know what name the layer accepted smart filter has. What the action knows is moving smart filter to first layer or second layer... is not moving smart filter to layer A or Layer B. 

The layers are in order. And I found them being like this. 
![Screenshot_20210317_113917_com](https://user-images.githubusercontent.com/79258245/111436433-f1d52f80-873c-11eb-8ad2-a47cfbdc0881.png)
![Screenshot_20210317_144416_com (3)](https://user-images.githubusercontent.com/79258245/111436467-fac60100-873c-11eb-9f49-cb4b2d6b5b28.png)

**Comment by nothingshiru** [comment]
The action "Clear smart filters" and "delete the first smart filter" should be different. 
"Hide smart filter" and "Show smart filter" are different too. 

I have made some actions similar but different and a PSD to test. 
[Actions.zip](https://github.com/photopea/photopea/files/6154885/Actions.zip)

**Comment by photopea** [comment]
I am trying to find out if there exists any document, where all possible actions are described in some form. Or if there exists a person, who knows it all and could explain it to me.

At the moment, my action processing was done "ad-hoc". I modified my code slightly for each new action I came across. Now, my code is quite messy. I am afraid that by adding support for new acitons, I could break the support for the old actions.
